### PR TITLE
switch to official Python installer on macOS + rework artifacts on Linux

### PR DESCRIPTION
**Pull request by benoit-pierre**
- no more brew on macOS: the official Python installer for 3.5 is used instead
- drop eggs and deb on Linux: instead generate an [AppImage](http://appimage.org/)
- add [packpack](https://github.com/packpack/packpack) support: this can be used to generate the AppImage, a deb for Ubuntu Trusty/Xenial, an Arch Linux package, or a RPM for Fedora Rawhide

### CODE REVIEW

**Comment by benoit-pierre** [comment]
Fixed the AppImage launcher (`LD_LIBRARY_PATH`)...

File: .travis.yml [hunk 1]
```diff
@@ -52,11 +52,17 @@ script:
...
   - $PYTHON setup.py patch_version
   - $PYTHON setup.py test
-
-before_deploy:
   - |
     (
     set -ex
+    # Only generate artifacts if we're actually going to deploy them.
+    # Note: if we moved this to the `before_deploy` phase, we would
+    # not have to check, but we'd also loose caching; since the
```
Comment (morinted) [inline .travis.yml]: should be "lose" caching

**Comment by morinted** [comment]
Reviewed, all looks good to me minus typo 👍

**Comment by benoit-pierre** [comment]
Fixed typo in `.travis.yml`.

**Pull request merged**